aten/src/ATen/native/native_functions.yaml

```diff
@@ -805,6 +805,17 @@
 - func: cosine_embedding_loss(Tensor input1, Tensor input2, Tensor target, float margin=0.0, int reduction=Mean) -> Tensor
   use_c10_dispatcher: full
 
+- func: count_nonzero.dim_IntList(Tensor self, int[] dim) -> Tensor
+  use_c10_dispatcher: full
+  variants: function, method
+
+- func: count_nonzero(Tensor self, int? dim=None) -> Tensor
+  use_c10_dispatcher: full
+  variants: function, method
+
+- func: count_nonzero.dim_DimnameList(Tensor self, Dimname[] dim) -> Tensor
```

Comment:
I would remove this variant, otherwise you'll have to add another test for it. It's not worth the hassle.

Reply:
Agreed. 
I was also having trouble with it. In Python, it wasn't correctly dispatching to the named variant when I tried to add test.

Have removed this as well as the mention in `namedinference.rst`